llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -590,27 +590,44 @@ void SpirvLowerGlobal::mapGlobalVariableToProxy(GlobalVariable *globalVar) {
   const auto &dataLayout = m_module->getDataLayout();
   Type *globalVarTy = globalVar->getValueType();
 
-  m_builder->SetInsertPointPastAllocas(m_entryPoint);
-
   Value *proxy = nullptr;
-
+  removeConstantExpr(m_context, globalVar);
   // Handle special globals, regular allocas will be removed by SROA pass.
-  if (globalVar->getName().startswith(RtName::HitAttribute))
+  if (globalVar->getName().startswith(RtName::HitAttribute)) {
     proxy = m_entryPoint->getArg(1);
-  else if (globalVar->getName().startswith(RtName::IncomingRayPayLoad))
+    globalVar->replaceAllUsesWith(proxy);
+  } else if (globalVar->getName().startswith(RtName::IncomingRayPayLoad)) {
     proxy = m_entryPoint->getArg(0);
-  else if (globalVar->getName().startswith(RtName::IncomingCallableData))
+    globalVar->replaceAllUsesWith(proxy);
+  } else if (globalVar->getName().startswith(RtName::IncomingCallableData)) {
     proxy = m_entryPoint->getArg(0);
-  else
-    proxy = m_builder->CreateAlloca(globalVarTy, dataLayout.getAllocaAddrSpace(), nullptr,
-                                    Twine(LlpcName::GlobalProxyPrefix) + globalVar->getName());
+    globalVar->replaceAllUsesWith(proxy);
+  } else {
+    // Collect used functions
+    SmallSet<Function *, 4> funcs;
+    for (User *user : globalVar->users()) {
+      auto inst = cast<Instruction>(user);
+      funcs.insert(inst->getParent()->getParent());
+    }
+    for (Function *func : funcs) {
+      assert(m_entryPoint);
```

Comment:
This assertion should maybe moved to the loop above, since you don't want to process anything once you found the first function?